internal/controllers/cluster/cluster_controller_phases_test.go

```diff
@@ -149,6 +211,202 @@ func TestClusterReconcilePhases(t *testing.T) {
 				} else {
 					g.Expect(err).ToNot(HaveOccurred())
 				}
+
+				if tt.check != nil {
+					tt.check(g, tt.cluster)
+				}
+			})
+		}
+	})
+
+	t.Run("reconcile control plane ref", func(t *testing.T) {
+		cluster := &clusterv1.Cluster{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "test-cluster",
+				Namespace: "test-namespace",
+			},
+			Status: clusterv1.ClusterStatus{
+				InfrastructureReady: true,
+			},
+			Spec: clusterv1.ClusterSpec{
+				ControlPlaneEndpoint: clusterv1.APIEndpoint{
+					Host: "1.2.3.4",
+					Port: 8443,
+				},
+				ControlPlaneRef: &corev1.ObjectReference{
+					APIVersion: "controlplane.cluster.x-k8s.io/v1beta1",
+					Kind:       "GenericControlPlane",
+					Name:       "test",
+				},
+			},
+		}
+		clusterNoEndpoint := &clusterv1.Cluster{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "test-cluster",
+				Namespace: "test-namespace",
+			},
+			Status: clusterv1.ClusterStatus{
+				InfrastructureReady: true,
+			},
+			Spec: clusterv1.ClusterSpec{
+				ControlPlaneRef: &corev1.ObjectReference{
+					APIVersion: "controlplane.cluster.x-k8s.io/v1beta1",
+					Kind:       "GenericControlPlane",
+					Name:       "test",
+				},
+			},
+		}
+
+		tests := []struct {
+			name         string
+			cluster      *clusterv1.Cluster
+			cpRef        map[string]interface{}
+			expectErr    bool
+			expectResult ctrl.Result
+			check        func(g *GomegaWithT, in *clusterv1.Cluster)
+		}{
+			{
+				name:      "returns no error if control plane ref is nil",
+				cluster:   &clusterv1.Cluster{ObjectMeta: metav1.ObjectMeta{Name: "test-cluster", Namespace: "test-namespace"}},
+				expectErr: false,
+			},
+			{
+				name:         "returns error if unable to reconcile control plane ref",
+				cluster:      cluster,
+				expectErr:    false,
+				expectResult: ctrl.Result{RequeueAfter: 30 * time.Second},
+			},
+			{
+				name:    "returns no error if infra config is marked for deletion",
+				cluster: cluster,
+				cpRef: map[string]interface{}{
+					"kind":       "GenericControlPlane",
+					"apiVersion": "controlplane.cluster.x-k8s.io/v1beta1",
+					"metadata": map[string]interface{}{
+						"name":              "test",
+						"namespace":         "test-namespace",
+						"deletionTimestamp": "sometime",
+					},
+				},
+				expectErr: false,
+			},
+			{
+				name:    "returns no error if infrastructure has the paused annotation",
```

Comment:
```suggestion
				name:    "returns no error if control plane has the paused annotation",
```
(Please also fix similar cases above & below)